src/mongo/MongoTripod.class.php

```diff
@@ -657,11 +633,12 @@ protected function getDataUpdater()
     {
         if(!isset($this->dataUpdater))
         {
+            $readPreference = $this->collection->getReadPreference();
             $opts = array(
                 'defaultContext'=>$this->defaultContext,
                 OP_ASYNC=>$this->async,
                 'stat'=>$this->stat,
-                'readPreference'=>$this->collection->getReadPreference(),
+                'readPreference'=>$readPreference['type'],
```

Comment:
@rsinger  I think we deal with it separately as a different issue.  There more important thing was whether our own $opts readPreference was always consistent and as far as I can tell it is.